### ISSUE TITLE
Current default scanline overlay configuration is not pixel perfect

### ISSUE THREAD

**Issue by gnadenlos** [ClusterM/retroarch-clover#90]
Recommended settings are:

Settings/Video:

- Aspect Ratio -> Custom 
- Aspect Ratio X Pos. -> 128 (identical pixel size, factor 4) or 192 (4:3) or 160 (Sega Genesis)
- Aspect Ratio Y Pos. -> 24
- Aspect Ratio Width -> 1024 (identical pixel size, factor 4) or 896 (4:3) or 960 (Sega Genesis)
- Aspect Ratio Height -> 672
- Integer Scale -> OFF
- Crop Overscan (Reload) -> ON
- Bilinear Filtering -> OFF

Settings/Onscreen Display/Onscreen Overlay:

- Display Overlay -> ON
- Overlay Preset -> scanlines.cfg 
- Overlay Opacity -> 0.70

That way it is pixel perfect. One scanline for each real pixel. 

**It is important to use "Crop Overscan = ON", so every SNES game will have the same vertical resolution of 224 pixels. That way using Aspect Ratio Height of 672 is factor 3 - this is the factor the overlay was made for.**

**Comment by gnadenlos** [comment]
Wrong section, this is a retroarch-clover issue.

**Comment by ClusterM** [comment]
You can tune it for yourself. It's a matter of taste.

**Comment by gnadenlos** [comment]
IMO it's a matter of taste if you use scanlines, if you combine them with other filters, etc. 

But if you use scanlines in form of an overlay you should always aim for one scanline per real pixel - that's how the overlay was designed and how the original description explained the configuration.

**Comment by ClusterM** [comment]
Ok.

**Comment by ClusterM** [comment]
But 75% opacity is too much.

Original canoe's canlines:

![image](https://user-images.githubusercontent.com/4236181/31424261-e888b79c-ae61-11e7-943b-c007d559c913.png)

RetroArch's 75% scanlines:

![image](https://user-images.githubusercontent.com/4236181/31424269-f2f24e3c-ae61-11e7-90c7-e4740342e8fe.png)
 
RetroArch's 50% scanlines:

![image](https://user-images.githubusercontent.com/4236181/31424277-faf967b4-ae61-11e7-9af8-7e07088f67f5.png)

RetroArch's 25% scanlines:

![image](https://user-images.githubusercontent.com/4236181/31424297-0f709ef6-ae62-11e7-9aa9-b2144cd57388.png)

**Comment by itends** [comment]
Is it necessary to have a shader active when we already use an overlay? Because right now the snes9x2010 shader is enabled. Or is it used for additional smoothing?

**Comment by ClusterM** [comment]
No, shaders sould be disabled.

**Comment by gnadenlos** [comment]
The opacity is a matter of taste, I just went for .70, because that was the recommendation in the image you linked on Twitter.

The really important options are:

Aspect Ratio Height -> 672 (or whatever corresponds to factor 3 for other systems)
Integer Scale -> OFF
Crop Overscan (Reload) -> ON

Because this will result in "pixel perfect" scanlines (224x3=672) for SNES and Genesis/MegaDrive. "Crop Overscan = ON" ensures, that all SNES games will output at 224 pixels.

The rest of the configuration is more flexible.

Using factor 3 (960 for Genesis) or 4 (1024 for SNES) for "Aspect Ratio Weight" is useful, because it will give you uniform pixel sizes. The problem is that you can't achieve a 4:3 aspect ratio using that values. That's why 896 pixels may be another option, that will give you 4:3 aspect ratio for SNES and Genesis.

**Comment by rsn8887** [comment]
Is it not possible to reproduce exactly the Clover CRT filter in Retroarch, by adjusting the aspect ratio, screen centering, filtering, and scanline appearance to match? 

I think that should be the goal, to have no discernible difference between Clover and Retroarch, for a uniform feel.

**Comment by rhester72** [comment]
Unfortunately, no.  It would require the use of shaders with 3x resampling, which when coupled with the Snex9x2010 core, the hardware just isn't capable of doing.  Using an overlay is a reasonable compromise - I've also been experimenting with using the NESCE shader with 2x resampling.  The scanlines look a bit better, but the "softness" it imparts to the image looks pretty odd and unappealing (to me).

**Comment by gromber** [comment]
i create a noScalines.cfg without scanlines.png for remain decorative frames without scanlines in overlay settings.

**Comment by gnadenlos** [comment]
The new scanlines in V1.0 are really close to Nintendos CRT mode, but you still are missing the important "Crop Overscan (Reload) -> ON" setting. 

Without that setting some games have a different height and you don't have factor 3 anymore.

Here is an example (Terranigma):

Using Crop Overscan = ON will result in factor 3:
https://i.imgur.com/mLlulZW.png

Without Crop Overscan = ON you don't get factor 3 for that game:
https://i.imgur.com/S7KMos3.png

Using "Crop Overcan = ON" will result in factor 3 for almost all games.

**Comment by rsn8887** [comment]
It is just not working for me. I have "half a scanline" on the top of the screen, and the scanlines are not perfectly "aligned" with the pixels, regardless of what I do. I tested with Secret of Evermore. Could it be that Secret of Evermore is one of those games where the default setting with "Crop Overscan ON" just does not work?

I think the default settings should either work for all games, or scanlines should be disabled by default. Scanlines look horrible if they are not aligned perfectly with the game pixels.

In addition, the default opacity on my screen is wrong, the scanlines are way too dark compared with CRT mode. The default overlay opacity should be 0.2, not 0.5, I think.

**Comment by gnadenlos** [comment]
@rsn8887: The current version has opacity 1.0 and a new overlay, so it seams you have some leftovers in your configuration if you still have 0.5 as default. You should try to use "unistall" in hakchi2 to get rid of those leftovers.

Are you talking about a NTSC version of the game with Retroarch? 

Here is my result:
https://i.imgur.com/I9LCY5K.jpg

**Comment by rsn8887** [comment]
@gnadenlos: You were right, I DID a leftover old overlay which did not get deleted with uninstall. I used FTP to update all files. But the new overlay seems to be off - it is not pixel perfect anymore now, even with Crop Overscan.

You can see it in your own example picture, the scanlines do not align with each pixel in the same way. I suspect some scaling is now wrong in the new overlay.

**Comment by gnadenlos** [comment]
@rsn8887: Yes, I prefer the old scanlines.png too, because it allows us to adjust the strength by setting a different opacity. 

I haven't verified the alignment of the new png, but if it is as you say, it should be fixed/replaced.

So for the moment we have to change the png and set Crop Overscan to ON ourselves, if we want pixel perfect scanlines. I hope Cluster will fix those problems, because many users don't know about the problem or don't know how to fix it themselves and will unnecessarily get subpar results.

**Comment by bankaiprimera** [comment]
@rsn8887: is it recommended to use the scanlines template you posted earlier with Hakchi 2.21f and the latest RA cores (https://github.com/KMFDManic/NESC-SNESC-Modifications/releases/), with opacity 1.00?

@gnadenlos and @rsn8887: What is currently the best way to enjoy pixel perfect scanlines?

thanks!

**Comment by rsn8887** [comment]
It is a matter of taste. I like my version (posted in a thread above), but the default is also ok.

In my version, the bright area of the scanlines are centered on the SNES pixels, with dark in-between.

In the version included with retroarch-clover currently, the bright area of the scanlines is in between the SNES pixels, e.g. the center of each pixel is dark.

You have to set opacity 100, that is intended. Otherwise the borders will also become transparent.

In order to install my version
- go to my earlier comment here (two posts above), 
- right-click on the scanline image in that comment
- select save linked file as
- save as scanlines.png
- transfer the file "scanlines.png" via ftp into the snes mini folder /etc/libretro/.config/retroarch/overlay/ . You can overwrite the original or rename it to "scanlines_orig.png" for example
Note: Ftp can be enabled on the snes mini while it is connected to windows. Then you can access the folders and copy files over.

**Comment by bankaiprimera** [comment]
@rsn8887 Thanks for the explanation. I can't find the .config folder, it doesn't exist.. /etc/libreto _does_ exist. Should I make a folder or something?

**Comment by NEStalgia** [comment]
Where can I download this overlay?
